### PR TITLE
e3sm-scorpio: add e3sm-scorpio package

### PR DESCRIPTION
**Pull request by downloadico**
This is the Scorpio package from the e3sm.org site.

### CODE REVIEW

File: var/spack/repos/builtin/packages/e3sm-scorpio/package.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    homepage = "https://e3sm.org/scorpio-parallel-io-library/"
+    url = "https://github.com/E3SM-Project/scorpio/archive/refs/tags/scorpio-v1.4.1.tar.gz"
+
+    version("1.4.1", sha256="7cb4589410080d7e547ef17ddabe68f749e6af019c1d0e6ee9f11554f3ff6b1a")
+
+    variant("timing", default="False", description="Enable timing")
+    variant("mpi", default="True", description="Enable MPI")
+
+    depends_on("cmake", type="build")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/e3sm-scorpio/package.py:22]: Is there a reason to not just rely on the `cmake` dependency inherited from `CMakePackage`?

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256 and ability to concretize the spec.  Just a minor question about the `CMake` dependency.

**Comment by tldahlgren** [comment]
Thank you for your contribution.  For future reference, it would simplify the review process if a PR-specific branch is used.  See https://spack.readthedocs.io/en/latest/contribution_guide.html#branching for the process.

File: var/spack/repos/builtin/packages/e3sm-scorpio/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/e3sm-scorpio/package.py:22]: no!  OOPS!

**Comment by downloadico** [comment]
https://github.com/spack/spack/pull/35794

**Pull request closed without merging**